### DISCUSSION TITLE
Sudden issue using Neurodesk in Google Colab

### DISCUSSION THREAD

**Discussion by marlapinkert** [neurodesk/neurodesk.github.io-hugo#458] [Q&A] [answered]
Dear all,

I am having issues using Neurodesk in Google Colab (Error message upon install: No module named 'lmod'). Subsequently, I assume this is an issue with lmod, but I have not been able to solve this yet. This has only started today - yesterday at 8 PM everything worked perfectly fine. 

My code is here:

```
import os
os.environ["LD_PRELOAD"] = "";
os.environ["APPTAINER_BINDPATH"] = "/content"
os.environ["LMOD_CMD"] = "/usr/share/lmod/lmod/libexec/lmod"

!curl -J -O https://raw.githubusercontent.com/NeuroDesk/neurocommand/main/googlecolab_setup.sh
!chmod +x googlecolab_setup.sh
!./googlecolab_setup.sh
os.environ["MODULEPATH"] = ':'.join(map(str, list(map(lambda x: os.path.join(os.path.abspath('/cvmfs/neurodesk.ardc.edu.au/neurodesk-modules/'), x),os.listdir('/cvmfs/neurodesk.ardc.edu.au/neurodesk-modules/')))))

import lmod
await lmod.load('fsl/6.0.4')
```

You can also see that it used to work very recently in these notebooks: https://github.com/marlapinkert/TEWA2/blob/main/04_Motion_Intro_Preprocessing.ipynb and https://github.com/marlapinkert/TEWA2/blob/main/05_Preprocessing_Nipype.ipynb. I am using the same code to install it as in the example provided (https://colab.research.google.com/drive/1g5cnZxj1llRaHmOs4xSglqsXnFkQYuol?usp=sharing). The example also does not work anymore. 

I am using Neurodesk in Colab for teaching and would prefer continuing with Colab as this is easiest for students. I would be very grateful for any help.

Best wishes,
Marla Pinkert

**Comment by stebo85** [comment]
Dear @marlapinkert - Thank you so much for flagging this issue! I can replicate the problem and we are investigating what happened - It looks like google colab updated something ...

**Comment by stebo85** [comment, marked as answer]
Dear @marlapinkert - we found the problem. jupyterlmod updated 20hours ago and broke the API: https://pypi.org/project/jupyterlmod/#history
https://github.com/cmd-ntrf/jupyter-lmod/pull/62/files#diff-2a139389071b29d80b4f6663679b8f1a774487d68e6f785ac50012525586b338

We implemented a workaround on our side, so it is working again.

**Reply by marlapinkert** [reply]
Thank you so much, this is amazing! I am very thankful (& impressed with how quickly you took care of it). :)

**Comment by stebo85** [comment]
Dear @marlapinkert - your notebooks look great :) Would you be keen to contribute them to our example repository? https://www.neurodesk.org/example-notebooks/intro.html

**Reply by marlapinkert** [reply]
Thank you, yes, I would love to! I will just remove the references I make to our lectures as they would be confusing to others. Is there anything else I should do? And do I just send them to you?  :)

**Reply by stebo85** [reply]
ideally you could send them as a pull request to the repository - then it's properly attributed to you: https://github.com/NeuroDesk/example-notebooks

We are also preparing an OHBM abstract about this and every contributor will be co-author :)
